### PR TITLE
Remove API subproject

### PR DESCRIPTION
**Pull request by mbj**
Depends on:

* [x] #664

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj there are still some references to the API project, like those in: https://github.com/MountainRoseHerbs/spree/blob/master/backend/app/assets/javascripts/spree/backend/payments/edit.js.coffee#L6

**Comment by mbj** [comment]
> @mbj there are still some references to the API project from spree_backend, like those in: https://github.com/MountainRoseHerbs/spree/blob/master/backend/app/assets/javascripts/spree/backend/payments/edit.js.coffee#L6

I'm aware of this, this branch is more to track this progress, I'll rebase it on top of each removal of references till this branch gets green.

File: .buildkite/hooks/post-checkout [hunk 1]
```diff
@@ -17,3 +18,6 @@ clean_output=$(
 
 # Test if the working directory is pristine
 [[ -z $clean_output ]] || (echo -e "Working directory not pristine:\n$clean_output"; exit 1)
+
+# Cleanup empty directories that are leftover from code removals
+git clean -d --force
```
Comment (dkubb) [inline .buildkite/hooks/post-checkout]: Without this change you were seeing errors from the other `git clean` call?

Should we make this change in the other repos?

Comment (dkubb) [inline .buildkite/hooks/post-checkout]: Doesn't this remove everything *including* directories? If so then you'll probably need to duplicate the `exclude` options as the command above.

**Comment by dkubb** [comment]
@mbj should we queue up a removal of the user "api key"? I would expect that it's tightly coupled to the api usage and not used elsewhere (?).

File: .buildkite/hooks/post-checkout (hunk 1 above)
Comment (mbj) [inline .buildkite/hooks/post-checkout]: good call, this commit is really old and I do not recall the details exactly let me reevaluate it.

**Comment by mbj** [comment]
> @mbj should we queue up a removal of the user "api key"? I would expect that it's tightly coupled to the api usage and not used elsewhere (?).

We use it in the main app for an export API.

File: .buildkite/hooks/post-checkout (hunk 1 above)
Comment (dkubb) [inline .buildkite/hooks/post-checkout]: Yeah if you can't determine a concrete reason for that commit I propose we remove it and then re-evaluate if the problem ever occurs necessitating it.

Comment (mbj) [inline .buildkite/hooks/post-checkout]: I do recall the details now after local experimentation:

The first `git clean` call uses `-x` to ignore the `.gitignore`, hence it needs `--exclude` options for the parts of the tree that are usually inside `.gitignore` that we do not consider as problematic if not pristine.

The 2nd `git clean` does not specify `-x` and thus does not need to re-specify the `--excludes`. And its needed so the empty directories do not leak to other (or the current builds).

Basically this change whitelists empty directories to not qualify as "dirty tree" but still removes them.

Comment (dkubb) [inline .buildkite/hooks/post-checkout]: It would probably result in the `buildkite-script-*` files being matched and removed, since they are not part of the `.gitignore`.

Comment (mbj) [inline .buildkite/hooks/post-checkout]: Mhh, why do these scripts exist there in the first place? And why does the build still pass if they are removed?

Comment (dkubb) [inline .buildkite/hooks/post-checkout]: They are added by buildkite. I don't know what their purpose is or if it is safe to remove them.

Perhaps they are added and then executed, and once executed are safe to remove, but I am not sure.

Comment (mbj) [inline .buildkite/hooks/post-checkout]: I'm fine to remove them given the build passes. Also its IMO better to have the repo in the exact state we'd deploy it. And not having "some dangling files" only present on CI.

Comment (dkubb) [inline .buildkite/hooks/post-checkout]: Sure, let's keep this as-is and worry about any potential issues later. Worst case a future build breaks and the fix will be trivial.

I'd bet part of the reason they are excluded is just to keep the output of the command minimized so it can be tested. In the case of this command we don't care at all about the output.
